src/http/query.c

```diff
@@ -43,11 +43,17 @@ Query* query_parser(char* buffer) {
 }
 
 void* query_get(const Query* restrict q, const char* restrict key) {
-	if(unlikely(!q || !key || *key))
-		return NULL;
+	if(unlikely(q == NULL))
+		return NULL;  // no query
+
+	if(unlikely(key == NULL))
+		return NULL;  // no key
+
+	if(unlikely(*key == '\0'))
+		return NULL;  // empty key
 
 	Node* cur = q->head;
-	while(cur && strcmp(SPAIR(cur)->key, key))
+	while(cur && strcmp(SPAIR(cur)->key, key) != 0)
```

Comment:
It is an unessential condition statement.
We know the strcmp's return value: https://man7.org/linux/man-pages/man3/strcmp.3.html

Reply:
Here should be like src/http/header.c line 107.
I forgot to change.